config/import-control.xml

```diff
@@ -7,7 +7,21 @@
     <allow pkg="java.io"/>
     <allow pkg="java.util"/>
     <allow pkg="com.github.checkstyle.regression.data"/>
+    <allow pkg="com.github.checkstyle.regression.util"/>
+    <allow pkg="org.immutables"/>
+
     <subpackage name="git">
         <allow pkg="org.eclipse.jgit"/>
     </subpackage>
+
+    <subpackage name="module">
+        <allow pkg="com.google.gson"/>
+        <!-- Keep this until we don't need hardcode ModuleInfo map. -->
+        <allow class="java.nio.charset.Charset"/>
+    </subpackage>
+
+    <subpackage name="util">
+        <allow pkg="com.google.gson"/>
```

Comment:
I'm not sure what happened to my original post on this as I can't find it anymore.

I am fine with `util` package, but it seems to me we are breaking component design by using component specific `com.google.gson` outside of `module` component, which is the only component that will use it right?
Should this be a `util` inside module?

It also seems like `ModuleUtils` is trying to do too much. It does json reading and other various things.
I think we should have one utility for JSON, including code in `GsonProvider`, and have another class that is what util module needs.